source/features/previous-version.tsx

```diff
@@ -0,0 +1,117 @@
+import * as pageDetect from 'github-url-detection';
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+
+import features from '../feature-manager';
+import * as api from '../github-helpers/api';
+import observe from '../helpers/selector-observer';
+
+import GitHubURL from '../github-helpers/github-url';
+
+const getHistoryOids = cache.function('previous-version', async (githubUrl: GitHubURL): Promise<string[] | false> => {
```

Comment:
It's best to accept location.href here and then call GitHubURL inside. This also makes it easier to handle cacheKey. 

The name could be `getPastCommits`